metadata/src/utils/retain.rs

```diff
@@ -356,4 +428,34 @@ mod tests {
             );
         }
     }
+
+    #[test]
+    fn issue_1659() {
+        let metadata_cache = load_metadata("../artifacts/regressions/1659.scale");
```

Comment:
Nit: rename to `full_metadata`?

Reply:
which `full_metadata`? maybe we can name it `1659.full_metadata.scale`?

Reply:
Sorry, I wasn't clear at all! I meant to rename `metadata_cache` to `full_metadata` :)